include/private/tdyconditionsimpl.h

```diff
@@ -22,6 +23,9 @@ typedef struct Conditions {
   /// Compute the pressure at a set of given points on the boundary.
   PetscErrorCode (*compute_boundary_pressure)(void*,PetscInt,PetscReal*,PetscReal*);
 
+  /// Set the pressure at a set of given points on the boundary.
+  PetscErrorCode (*assign_boundary_pressure_type)(void*,PetscInt,PetscReal*,PetscInt*);
```

Comment:
I guess I still don't quite understand why we need this. Does the two-point flux approximation actually support Neumann pressure boundary conditions in addition to the Dirichlet/seepage types? Wouldn't a "Neumann" pressure boundary condition be related linearly to a boundary condition on the normal velocity? If so, we could check the `compute_boundary_pressure` and `compute_boundary_velocity` function pointers to determine the type of boundary condition. Specifically:

* If `compute_boundary_pressure` and `compute_boundary_velocity` are both `NULL`, we have a seepage BC.
* If `compute_boundary_pressure` is non-`NULL`, we have a Dirichlet pressure BC.
* If `compute_boundary_velocity` is non-`NULL`, we have a Neumann pressure BC.

Might that be simpler than adding a different function type? If needed we could add a helper function to check these pointers and return something that indicates the "pressure BC type". But I very well could be missing something here.

Reply:
- If no BC is specified on a face, it will be a Neumann with zero-flux. (Currently, the Neumann BC implies a zero-flux. Eventually, an non zero-flux Neumann BC needs to be supported.)
- This is indeed a temporary way of adding dirichlet and seepage BC in the domain. 
- After your changes are merged, I envision there would be support for a user-defined function that adds a BC condition: (1) BC-type and (2) values of the BC on a collection of faces.